### PR TITLE
Fix inbound transfer to obey warnings

### LINKED ISSUE #36627: Domains: Possible to reach domain transfer options for unsupported TLDs
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/domains/add/transfer/{SITE_URL}
2. Type in a domain with a TLD not supported for transfer
3. Click "Transfer". An error message will appear that the TLD is not supported for transfer
4. Click "Transfer" again. Optionally, make a change to the domain searched before clicking Transfer a second time.

#### What I expected
To still see the error that the TLD is unsupported.

#### What happened instead
I'm taken to the next screen that checks if the domain is unlocked, and asking for the Auth code.

This happens with unsupported TLDs we're accredited for as a registrar, and ones that we aren't.

#### Screenshot / Video
domain

#### Context / Source

Did some testing after two conversations in three days about users apparently able to reach transfer options for unsupported TLDs:

p1570458789063600-slack-domains
p1570611103015100-slack-domains

And one on users unable to reach transfer options for a TLD that is supported (since fixed, but might be relevant given the timing):
p2MSmN-7v4-p2

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* Fix our inbound transfer logic to ensure that any notices are treated as blocking
* Fix the Back button when transferring a domain for an existing site

#### Testing instructions

There are two main flows to test here:
1. Transferring a domain during signup/site creation, and
2. Transferring a domain for an existing site.

**Site Creation**
* Start from `/start` and move through the signup flow.
* Work through the flow until you get to the domain search UI at `/start/domains-with-preview`.
* Search for a domain (the actual domain doesn't matter).
* When the results appear, scroll down to the "Already own a domain?" row and click on it.
* On the next page, click on the "Transfer to WordPress.com" button (which should be in the left card).
* On the next page, type in the following types of domains, click the "Transfer" button, and verify that you get the expected behaviour in each case:
  * Use a domain that is already registered *and* is in a TLD which we don't support transfers for, such as `cbc.ca` or `governo.it`. The following should occur:
    * The signup flow *does not* proceed to plan selection. (Note that this may occur for an available domain where we support registering the domain -- `.ca` is such a TLD.)
    * We display a notice for the TLD like: "We don't support transfers for domains ending with **.ca**, but you can map it instead."
    * If you click on "Transfer" again, you get the same notice and remain on the same page.
  * Use a domain that is available *and* is in a TLD that we support registrations for, such as `sometestsite97531.ca`. The following should occur:
    * You should be moved to the plan selection page, as we've switched to registering the domain.
  * Use a domain that is available *and* is in a TLD that we *don't* support registrations for, such as `sometestsite97531.it`. The following should occur:
    * The signup flow _should not proceed_ to plan selection.
    * We display a notice for the TLD like: "This domain is available to be registered, but we don't support transfers for domains ending with **.it**. If you register it elsewhere, you can map it instead."

**Adding a domain to a site**
* Start from a site you manage and go to *Manage* > *Domains*.
* Click on the "Add Domain" button.
* Scroll down to the "Already own a domain?" row and click on it.
* On the next page, click on the "Transfer to WordPress.com" button (which should be in the left card).
* Click on the "Back" link in the top left corner, and verify that you are returned to the screen above.
* click on the "Transfer to WordPress.com" button again.
* On the next page, type in the following types of domains, click the "Transfer" button, and verify that you get the expected behaviour in each case:
  * Use a domain that is already registered *and* is in a TLD which we don't support transfers for, such as `cbc.ca` or `governo.it`. The following should occur:
    * The page should not proceed to the transfer precheck page.
    * We display a notice for the TLD like: "We don't support transfers for domains ending with **.ca**, but you can map it instead."
    * If you click on "Transfer" again, you get the same notice and remain on the same page.
  * Use a domain that is available *and* is in a TLD that we support registrations for, such as `sometestsite97531.ca`. The following should occur:
    * The page should not proceed to the transfer precheck page.
    * An upgrade section/nudge should display for the domain with core text like "sometestsite97531.ca is available!". You should be able to click on the "Upgrade" or "Select" button and see checkout. ("Upgrade" is visible if you're on a free site; "Select" is visible if you're on a site with a plan.)
  * Use a domain that is available *and* is in a TLD that we *don't* support registrations for, such as `sometestsite97531.it`. The following should occur:
    * The page should not proceed to the transfer precheck page.
    * We display a notice for the TLD like: "This domain is available to be registered, but we don't support transfers for domains ending with **.it**. If you register it elsewhere, you can map it instead."

Fixes #36627

### CODE REVIEW

**Comment by delputnam** [comment]
Thanks for the amazing testing instructions, Dale! 💙

**Comment by daledupreez** [comment]
@klimeryk and @delputnam, I think a *lot* of the naming in the state is poor! However, I didn't want to take on a wholesale refactor just because I felt like it.

Does it make sense to limit my changes to a cleaner `isTransferrable` property of the state? Or should I look to refactor more of the component?

**Comment by daledupreez** [comment]
Side note: the e2e tests that failed were disabled earlier today, so I didn't break those with this PR. 😁

**Comment by delputnam** [comment]
> Does it make sense to limit my changes to a cleaner `isTransferrable` property of the state? Or should I look to refactor more of the component?

I'd vote to limit this PR to the changes needed to address the primary issue along any small-ish common-sense refactors that are limited to the parts of the code you touch. I always think it's ok to make sensible updates when you "crack open" a component. But I'd say that a wholesale refactor should be in a separate PR.

I think that the way you handled this PR was perfect (ignoring the ugly bits but fixing the issue)...my only hesitation was using `notice` for something that it wasn't exactly intended for since I worrying that it might cause problems in the future if someone changes how notices work.

That's not a very definitive answer.  I'm sure @klimeryk will have better elucidated thoughts on the matter. :-D

**Comment by daledupreez** [comment]
The big problem here is that it's not just `notice`, the code is *also* using `suggestion` to capture that we've found an available domain that should be treated as a suggestion for an upgrade! 🙈And then it's using `precheck` to indicate that we have a domain that we can transfer in and should show the pre-transfer checklist... 😦🙈
I *definitely* stopped myself!

**Comment by delputnam** [comment]
But I think that `suggestion` only represents one thing....and it's either empty or not.  And `precheck` is either falsy or true and indicates that we should show the pre-check component...maybe I'm misunderstanding though.

**Comment by daledupreez** [comment]
@delputnam, they do only represent one thing, but for me the naming is a bit terse and doesn't make it easy for me to actually understand what they represent. The suggestion is a purchase suggestion for an available domain. The pre-check component is shown when the requested domain can actually be transferred in, but having a simple precheck flag doesn't really make those requirements obvious.

TLDR; I found it hard to work out what on earth was actually happening in this component because it involves two environments, and has fairly complex logic that needs deep understanding of all the moving pieces to track well. I think this could be simplified a bunch with better naming along with some refactoring.

**Comment by delputnam** [comment]
@daledupreez I agree that we've got some poor naming going on here.  For the moment, let's just take care of the use of `notice` as I think that is the one that could potentially cause problems if the way that notices are implemented gets changed for some reason.  Then we can get this one merged. :-)
